pkg/system/gateway/factory.go

```diff
@@ -298,23 +301,23 @@ func (s *System) announceControllerNode(ctx context.Context, hubNode *gen.HubNod
 	})
 }
 
-// announceProxyNode sets up proxying for a node that is also a proxy.
-// This is similar to [announceControllerNode] but we skip updating the routing table as we assume all proxies have the same table and
-// including the other proxies table in ours would be redundant (and possible cause infinite routing loops).
+// announceProxyNode sets up proxying for a node that is also a gateway.
+// This is similar to [announceControllerNode] but we skip updating the routing table as we assume all gateways have the same table and
+// including the other gateways' table in ours would be redundant (and possible cause infinite routing loops).
 func (s *System) announceProxyNode(ctx context.Context, hubNode *gen.HubNode, nodeConn *grpc.ClientConn) {
 	go s.retry(ctx, "proxyNodeParent", func(ctx context.Context) (task.Next, error) {
 		return s.announceNodeParent(ctx, nodeConn, hubNode.Name)
 	})
-	// explicitly don't fetch proxy children as they will have the same children as us anyway
+	// explicitly don't fetch gateway children as they will have the same children as us anyway
 
 	// proxy any non-trait apis that also use routing
 	go s.retry(ctx, "proxyNodeApis", func(ctx context.Context) (task.Next, error) {
 		return s.announceNodeApis(ctx, hubNode, nodeConn)
 	})
 }
 
-// isProxy discovers if nodeConn is a proxy node, a node that has an enabled proxy system.
-func (s *System) isProxy(ctx context.Context, nodeConn *grpc.ClientConn) (bool, error) {
+// isGateway discovers if nodeConn is a proxy node, a node that has an enabled proxy system.
```

Comment:
couple of missed renames here

Reply:
Also I think you need to change the logic of this function to also check for the legacy name too.